src/main/java/org/sagebionetworks/web/client/widget/entity/EntityMetadata.java

```diff
@@ -58,7 +61,8 @@ public EntityMetadata(
     SynapseJSNIUtils jsni,
     RestrictionWidget restrictionWidgetV2,
     ContainerItemCountWidget containerItemCountWidget,
-    PortalGinInjector ginInjector
+    PortalGinInjector ginInjector,
+    EntityModalWidget entityModalWidget
```

Comment:
Instead of this widget and its view directly rendering the React component, refactor so that the React component is wrapped in and abstracted by this `EntityModalWidget`.